### PR TITLE
#1506 - In Store pickup shipping method support

### PR DESCRIPTION
**Pull request by IvansZuks**
GitHub issue: https://github.com/scandipwa/scandipwa/issues/1506

### CODE REVIEW

**Comment by lianastaskevica** [comment]
related BE PR - https://github.com/scandipwa/quote-graphql/pull/65

File: packages/scandipwa/src/component/CheckoutDeliveryOptions/CheckoutDeliveryOptions.component.js [hunk 1]
```diff
@@ -86,13 +92,28 @@ export class CheckoutDeliveryOptions extends PureComponent {
...
                     { this.renderShippingMethods() }
                 </ul>
                 { this.renderSelectedShippingMethod() }
+                <StoreInPickUp
+                  estimateAddress={ estimateAddress }
+                  shippingMethods={ shippingMethods }
+                  onStoreSelect={ onStoreSelect }
+                  onShippingMethodSelect={ onShippingMethodSelect }
+                  setSelectedShippingMethodCode={ setSelectedShippingMethodCode }
+                />
```
Comment (alfredsgenkins) [inline packages/scandipwa/src/component/CheckoutDeliveryOptions/CheckoutDeliveryOptions.component.js:116]: Is really always neccessary, cannot it just be there when needed? I mean in `renderSelectedShippingMethod` ?

File: packages/scandipwa/src/component/CheckoutDeliveryOptions/CheckoutDeliveryOptions.container.js [hunk 2]
```diff
@@ -40,14 +46,20 @@ export class CheckoutDeliveryOptionsContainer extends PureComponent {
...
         const result = shippingMethods.find(
-            ({ method_code, carrier_code }) => `${carrier_code}_${method_code}` === shippingMethod
-        ) || shippingMethods[0] || {};
+            ({ method_code, carrier_code }) => (`${carrier_code}_${method_code}` === shippingMethod
+            ) && method_code !== STORE_IN_PICK_UP_METHOD_CODE
+        ) || (
+            shippingMethods[0] && shippingMethods[0].method_code !== STORE_IN_PICK_UP_METHOD_CODE
+                ? shippingMethods[0]
+                : {}
+        );
```
Comment (alfredsgenkins) [inline packages/scandipwa/src/component/CheckoutDeliveryOptions/CheckoutDeliveryOptions.container.js:55]: Add some comments to this please, I do not understand what is going one here :)

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js [hunk 3]
```diff
@@ -46,18 +50,53 @@ export class CheckoutShippingContainer extends PureComponent {
...
+            telephone: phone,
+            street: [street],
+            firstname: name,
+            lastname: 'Store',
+            extension_attributes: [
+                {
+                    attribute_code: STORE_IN_PICK_UP_ATTRIBUTE_CODE,
+                    value: pickup_location_code
+                }
+            ]
```
Comment (alfredsgenkins) [inline packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js:99]: Liana says there is an option to disable this method. Should this really be there like... always?

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]
I do not liek then people from core team disable eslint rules... Please fix!

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js (hunk 3 above)
Comment (IvansZuks) [reply packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js:99]: There is the check in method `onShippingSuccess` is it there or not:
image

File: packages/scandipwa/src/component/CheckoutDeliveryOptions/CheckoutDeliveryOptions.container.js (hunk 2 above)
Comment (IvansZuks) [reply packages/scandipwa/src/component/CheckoutDeliveryOptions/CheckoutDeliveryOptions.container.js:61]: Added

**Review by alfredsgenkins — APPROVED** [review]

**Pull request merged**